okio/src/jvmMain/kotlin/okio/Timeout.kt

```diff
@@ -32,6 +33,12 @@ actual open class Timeout {
   private var deadlineNanoTime = 0L
   private var timeoutNanos = 0L
 
+  /**
+   * A sentinel that is updated to a new object on each call to [cancel]. Sample this property
+   * before and after an operation to test if the timeout was canceled during the operation.
+   */
+  @Volatile private var cancelMark: Any? = null
```

Comment:
I also considered doing `cancelCount: Int` but I think I like this better. It doesn’t have an overflow problem!

Reply:
I think it would have worked. Concurrent read, add, writes would still have changed the value. Overflowing still results in a new value. The only problem would have been if you called cancel 2^32 times during an operation it wouldn't be detected (but that's probably okay because it's impossible).

Reply:
That’s exactly the case I was considering. It’s practically impossible but ugly